### ISSUE TITLE
False E0107 diagnostic

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#21816]
```rust

pub trait Lock<T> {
    fn new(b: T) -> Self;
}
pub trait LockChoice {
    type Lock<T>: Lock<T>;
}
fn f<L: LockChoice>() {
    <L as LockChoice>::Lock::new(());
}
```
r-a errors on `<L as LockChoice>::Lock::new(())` with "this type alias takes 1 generic argument but 0 generic arguments were supplied "
